Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.psm1

```diff
@@ -0,0 +1,559 @@
+            {
+                $paramsToRemove += $paramName
+            }
+        }
+
+        foreach ($paramName in $paramsToRemove)
+        {
+            $result.Remove($paramName)
+        }
+
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-O365DscHashtableToString -Hashtable $result)"
+        return $result
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Policy,
+
+        [Parameter()]
+        [ValidateSet("InOrganization","NotInOrganization","None")]
+        [System.String[]]
+        $AccessScope,
+
+        [Parameter()]
+        [System.Boolean]
+        $BlockAccess,
+
+        [Parameter()]
+        [ValidateSet("All", "PerUser")]
+        [System.String]
+        $BlockAccessScope,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $ContentContainsSensitiveInformation,
+
+        [Parameter()]
+        [System.String[]]
+        $ContentPropertyContainsWords,
+
+        [Parameter()]
+        [System.Boolean]
+        $Disabled,
+
+        [Parameter()]
+        [System.String[]]
+        $GenerateAlert,
+
+        [Parameter()]
+        [System.String[]]
+        $GenerateIncidentReport,
+
+        [Parameter()]
+        [ValidateSet("All", "Default", "Detections", "DocumentAuthor", "DocumentLastModifier", "MatchedItem", "RulesMatched", "Service", "Severity", "Title")]
+        [System.String[]]
+        $IncidentReportContent,
+
+        [Parameter()]
+        [ValidateSet("FalsePositive", "WithoutJustification", "WithJustification")]
+        [System.String[]]
+        $NotifyAllowOverride,
+
+        [Parameter()]
+        [System.String]
+        $NotifyEmailCustomText,
+
+        [Parameter()]
+        [System.String]
+        $NotifyPolicyTipCustomText,
+
+        [Parameter()]
+        [System.String[]]
+        $NotifyUser,
+
+        [Parameter()]
+        [ValidateSet("Low", "Medium", "High", "None")]
+        [System.String]
+        $ReportSeverityLevel,
+
+        [Parameter()]
+        [ValidateSet("Ignore", "RetryThenBlock")]
+        [System.String]
+        $RuleErrorAction,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting configuration of DLPComplianceRule for $Name"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+                      -Platform SecurityComplianceCenter
+
+    $CurrentRule = Get-TargetResource @PSBoundParameters
+
+    if (('Present' -eq $Ensure) -and ('Absent' -eq $CurrentRule.Ensure))
```

Comment:
Is there a specific reason that you placed the text on the left of the comparison? I know this is a best practice for a comparison with $null, but is this also true for other comparisons?